odoo/osv/expression.py

```diff
@@ -1039,8 +1039,8 @@ def push_result(leaf):
 
                 else:
                     # rewrite condition to match records with/without relations
-                    op1 = 'inselect' if operator in NEGATIVE_TERM_OPERATORS else 'not inselect'
-                    subquery = 'SELECT "%s" FROM "%s" where "%s" is not null' % (rel_id1, rel_table, rel_id1)
+                    op1 = 'exists' if operator in NEGATIVE_TERM_OPERATORS else 'not exists'
+                    subquery = 'SELECT 1 FROM "%s" where "%s"."%s" = "%s".id' % (rel_table, rel_table, rel_id1, leaf.generate_alias())
```

Comment:
If `not exists` is optimised in PostgreSQL, I'm not aware of such optimisation on the `exists` use case.

Reply:
Yes, but it will create code complexity if we want `not exist` (when negative) and `in` (when positive) in the same version.

As far as I can tell, `IN` and `EXISTS` will lead to the same plan (in this case), right ? If yes, let be consistent as most at possible, no ?

Reply:
We test it with @rco-odoo and large datasets, and `exists` produce decent plan.